python/ray/serve/application.py

```diff
@@ -0,0 +1,190 @@
+from typing import List, Dict, TextIO, Optional, Union
+import json
+import yaml
+import time
+
+import ray
+from ray import serve
+from ray.serve.api import (
+    Deployment,
+    _get_global_client,
+)
+from ray.serve.common import DeploymentStatus
+from ray.dashboard.modules.serve.schema import (
+    ServeApplicationSchema,
+    serve_application_to_schema,
+    schema_to_serve_application,
+    serve_application_status_to_schema,
+)
+from ray.dashboard.modules.serve.sdk import ServeSubmissionClient
+from ray.serve.utils import logger
+
+
+class Application:
+    def __init__(self, deployments: List[Deployment]):
+        self._deployments = dict()
+        for idx, d in enumerate(deployments):
+            if isinstance(d, Deployment):
+                self._deployments[d.name] = d
+            else:
+                raise TypeError(
+                    f"Got object of type {type(d)} at index "
+                    f"{idx}. Expected only Deployment objects to "
+                    "be passed in."
+                )
+
+    def add_deployment(self, deployment: Deployment):
+        """Add deployment to the list. Validate name uniqueness."""
+
+        if not isinstance(deployment, Deployment):
+            raise TypeError(f"Got {type(deployment)}. Expected deployment.")
+
+        self._deployments[deployment.name] = deployment
+
+    def deploy(self, blocking: bool = True):
+        """Async deploy."""
```

Comment:
can we expand this docstring and make it clear the use of "blocking" and what the expected behavior should be ? 

Generally speaking it's recommended and valuable to use google's python style guide for critical functions, specially ones that are end user facing.
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Same applies to other functions as well as class level docstring. This will be a highly visible component so please polish it with a lot of care and good examples :)

Reply:
Good idea– I added docstrings to all the functions and the class.